manifests/job/present.pp

```diff
@@ -14,19 +17,51 @@
 #     if the job should be enabled
 #
 define jenkins::job::present(
-  $config,
+  $config = undef,
+  $temp_config_path = undef,
   $jobname  = $title,
   $enabled  = 1,
 ){
   include jenkins::cli
   include jenkins::cli::reload
 
+  if $temp_config_path != undef and $config != undef {
+    fail('You cannot set both \$temp_config_path AND $config param, only one is required')
+  }
+
   if $jenkins::service_ensure == 'stopped' or $jenkins::service_ensure == false {
     fail('Management of Jenkins jobs requires \$jenkins::service_ensure to be set to \'running\'')
   }
 
   $jenkins_cli        = $jenkins::cli::cmd
-  $tmp_config_path    = "/tmp/${jobname}-config.xml"
+  if $config == undef {
```

Comment:
I'd like to add an `if` statement here to test that we have either `$config` or `$temp_config_path` set but it makes the rpsec tests fail. Put a similar comment in #422